workers/notifier/src/main/kotlin/notifier/NotifierRunner.kt

```diff
@@ -70,17 +70,36 @@ class NotifierRunner {
                 workerContext.configManager.getSecret(Path(it))
             }
 
-            val notifierConfiguration = if (mailServerUser != null && mailServerPassword != null) {
-                NotifierConfiguration(
-                    mail = config.mail?.mailServerConfiguration?.copy(
+            val sendMailConfiguration = if (mailServerUser != null && mailServerPassword != null) {
+                config.mail?.mailServerConfiguration?.copy(
                         username = mailServerUser,
                         password = mailServerPassword
                     )?.mapToOrt()
-                )
             } else {
-                NotifierConfiguration()
+                null
+            }
+
+            val jiraRestClientUsername = config.jira?.jiraRestClientConfiguration?.username?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+            val jiraRestClientPassword = config.jira?.jiraRestClientConfiguration?.password?.let {
+                workerContext.configManager.getSecret(Path(it))
+            }
+
+            val jiraConfiguration = if (jiraRestClientUsername != null && jiraRestClientPassword != null) {
```

Comment:
Nit: You could drop the `else { null }` part by replacing the `if` construct with `takeIf`, e.g. something like

```
val jiraConfiguration = config.jira?.jiraRestClientConfiguration?.takeIf { 
  jiraRestClientUsername != null && jiraRestClientPassword != null
}?.copy(username = jiraRestClientUsername, password = jiraRestClientPassword)?.mapToOrt()
```